app/controllers/downloads_controller.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+# DownloadsController: for handling user downloads
+class DownloadsController < ApplicationController
+  before_action :authenticate_user!
+
+  def download
+    redirect_to root_url unless current_user.can_export?
```

Comment:
also include error handling for invalid and expired links?

![Screen Shot 2020-06-10 at 9 58 58 PM](https://user-images.githubusercontent.com/9956561/84336620-a797a000-ab65-11ea-8fda-1789f11d902a.png)

Reply:
Yeah, I agree that might be useful. Right now it simply returns a HTTP status (in this case a 404), but at minimum I'll include a message with an explanation.

Reply:
Done!